bin/custom/concerttours/src/concerttours/facades/impl/DefaultBandFacade.java

```diff
@@ -0,0 +1,66 @@
+package concerttours.facades.impl;
+
+import concerttours.data.BandData;
+import concerttours.data.TourSummaryData;
+import concerttours.facades.BandFacade;
+import concerttours.model.BandModel;
+import concerttours.service.BandService;
+import de.hybris.platform.core.model.product.ProductModel;
+import de.hybris.platform.servicelayer.dto.converter.Converter;
+import org.apache.commons.collections4.CollectionUtils;
+import org.springframework.beans.factory.annotation.Required;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DefaultBandFacade implements BandFacade {
+
+    private BandService bandService;
+    private Converter<BandModel, BandData> bandConverter;
+    private Converter<ProductModel, TourSummaryData> tourSummaryConverter;
+
+    @Override
+    public BandData getBand(String id) throws IllegalArgumentException {
+        if (id == null) {
```

Comment:
Please consider using Objects.requireNonNull instead of manual checks
(or get rid of checks like this one at all, it will anyway throw a descriptive NPE when null is passed)